green/result.py

```diff
@@ -302,29 +305,32 @@ def __getstate__(self):
             result_dict[pickle_attr] = self.__dict__[pickle_attr]
         return result_dict
 
-    def __setstate__(self, dict):
+    def __setstate__(self, state: dict[str, Any]) -> None:
         """
-        Since the callback functions weren't pickled, we need to init them
+        Since the callback functions weren't pickled, we need to initialize them.
         """
-        self.__dict__.update(dict)
+        self.__dict__.update(state)
         self.start_callback = None
         self.finalize_callback = None
 
-    def startTest(self, test):
+    def startTest(self, test: TestCaseT):
         """
-        Called before each test runs
+        Called before each test runs.
         """
         test = proto_test(test)
         self.start_time = time.time()
         self.reinitialize()
         if self.start_callback:
             self.start_callback(test)
 
-    def stopTest(self, test):
+    def stopTest(self, test: TestCaseT):
         """
-        Called after each test runs
+        Called after each test runs.
         """
-        self.test_time = str(time.time() - self.start_time)
+        if self.start_time:
+            self.test_time = str(time.time() - self.start_time)
+        else:
+            self.test_time = "0.0"
```

Comment:
This is specifically for skipped tests that do not have a start time but they have `stopTest` called on them in python 3.12.1